### PR TITLE
Upgrade to spring boot 3.1.10

### PR DESCRIPTION
**Pull request by kunalvarpe**
Reason:
- Nats should support spring boot 3.x.

Fix: gh-42

### CODE REVIEW

**Comment by kunalvarpe** [comment]
@scottf These are initial changes. I am not able to run full clean install due to gpg keys issues on my laptop. Please have a initial look at the PR.

**Comment by YunaBraska** [comment]
@kunalvarpe you can switch the java versions in the [travis.yaml](https://github.com/nats-io/spring-nats/blob/main/.travis.yml) for the CI to pass. Probably now it's every java version from 17 on.

**Comment by kunalvarpe** [comment]
> @kunalvarpe you can switch the java versions in the [travis.yaml](https://github.com/nats-io/spring-nats/blob/main/.travis.yml) for the CI to pass. Probably now it's every java version from 17 on.

Yes, you are right everything would be java 17 as Spring Boot 3.x has baseline of Java 17.

I am doing few changes in my initial commit. I will incorporate CI changes also. Thank you for the pointer.

**Comment by kunalvarpe** [comment]
@YunaBraska @scottf I have upgraded the `travis.yml` file with latest jdk version. Only jdk17 build succeeded all other failed due to 'unable to locate the jdk distribution on the ubuntu image'. Can someone help me here?

File: nats-spring-samples/multi-connect-sample/src/main/java/io/nats/spring/MultiConnectTransformer.java [hunk 1]
```diff
@@ -16,33 +16,38 @@
...
-            for (int i = value.length() - 1; i >= 0; i--) {
-                reverse.append(value.charAt(i));
-            }
-            message = reverse.toString().getBytes(StandardCharsets.UTF_8);
-        } else if (message instanceof String) {
-            String value = (String) message;
-            StringBuilder reverse = new StringBuilder();
-            for (int i = value.length() - 1; i >= 0; i--) {
-                reverse.append(value.charAt(i));
+    private static final Logger LOG = LoggerFactory.getLogger(MultiConnectTransformer.class);
```
Comment (YunaBraska) [inline nats-spring-samples/multi-connect-sample/src/main/java/io/nats/spring/MultiConnectTransformer.java:32]: I bereave we need to get rid of logging things before merge ;)

Comment (kunalvarpe) [reply nats-spring-samples/multi-connect-sample/src/main/java/io/nats/spring/MultiConnectTransformer.java:32]: @YunaBraska So are we thinking on getting rid of logging from samples projects?

Comment (YunaBraska) [reply nats-spring-samples/multi-connect-sample/src/main/java/io/nats/spring/MultiConnectTransformer.java:32]: ah thats a sample, then never mind :D

Comment (kunalvarpe) [reply nats-spring-samples/multi-connect-sample/src/main/java/io/nats/spring/MultiConnectTransformer.java:32]: Cool 😄

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 language: java
 sudo: required
 jdk:
-  - oraclejdk8
-  - oraclejdk9
-  - openjdk8
-  - openjdk10
-  - openjdk11
+  - openjdk17
+  - openjdk18
```
Comment (YunaBraska) [inline .travis.yml:6]: @scottf can you help here to get the other JDKs to work? I also couldn't find any list for travis JDK names :/

Comment (kunalvarpe) [reply .travis.yml:6]: @YunaBraska This is fixed with Jammy dist for Travis JDKs. You can see latest build succeeded for all JDKS from Java17 to Java22.

**Review by YunaBraska — APPROVED** [review]

**Comment by kis024** [comment]
Hello 2 all, any progress with this PR?

**Comment by scottf** [comment]
Sorry for the delay, been dealing with customers, will look at this next week.

**Comment by kunalvarpe** [comment]
@scottf Are we planning any release date for this?

**Comment by scottf** [comment]
@kunalvarpe @YunaBraska Do we need to semver major version for this? Otherwise I will just do a minor

**Comment by kunalvarpe** [comment]
> @kunalvarpe @YunaBraska Do we need to semver major version for this? Otherwise I will just do a minor

I believe it should be major version as underlying spring cloud version bump is also major. @YunaBraska What is your take on this?

**Comment by scottf** [comment]
@kunalvarpe I'm good with the major version. Can you add that to this PR?

**Comment by kruchkou** [comment]
Hello @kunalvarpe,
Could you provide any estimations of when a version for Boot 3 will be released?

**Comment by scottf** [comment]
@kruchkou Releasing 0.6.0-3.1

**Pull request merged**